pkg/structs/metrics.go

```diff
@@ -0,0 +1,33 @@
+package structs
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type PrometheusObserver interface {
+	WithLabelValues(lvs ...string) prometheus.Observer
+}
+
+type MetricGroup map[string]time.Duration
+
+func (mg MetricGroup) Observe(name string, dur time.Duration) {
+	mg[name] = dur
+}
+
+func (mg MetricGroup) ObserveSince(name string, t time.Time) {
+	mg[name] = time.Since(t)
+}
+
+func (mg MetricGroup) Commit(t PrometheusObserver, name string) {
+	for metric, dur := range mg {
+		t.WithLabelValues(name, metric, "").Observe(float64(dur.Milliseconds() / 1_000))
```

Comment:
that would be wrong. the float second gives you like 0.5 second and on, it's not like 0  for anything below 1

Reply:
_wrong_ -> _not the optimal way_, since the code would work 😉

But yes, thank you, I missed that! Changed 👍